tests/regression_tests/unstructured_mesh/source_sampling/test.py

```diff
@@ -0,0 +1,190 @@
+for i, (lib, schemes) in enumerate(product(*param_values)):
+    test_cases.append({'library' : lib,
+                       'schemes' : schemes,
+                       'inputs_true' : 'inputs_true{}.dat'.format(i)})
+
+
+@pytest.mark.parametrize("test_opts", test_cases)
+def test_unstructured_mesh(test_opts):
+
+    openmc.reset_auto_ids()
+
+    # skip the test if the library is not enabled
+    if test_opts['library'] == 'moab' and not openmc.lib._dagmc_enabled():
+        pytest.skip("DAGMC (and MOAB) mesh not enabled in this build.")
+
+    if test_opts['library'] == 'libmesh' and not openmc.lib._libmesh_enabled():
+        pytest.skip("LibMesh is not enabled in this build.")
+
+    ### Materials ###
+    materials = openmc.Materials()
+
+    water_mat = openmc.Material(material_id=3, name="water")
+    water_mat.add_nuclide("H1", 2.0)
+    water_mat.add_nuclide("O16", 1.0)
+    water_mat.set_density("atom/b-cm", 0.07416)
+    materials.append(water_mat)
+
+    materials.export_to_xml()
+
+    ### Geometry ###
+    dimen = 10
+    size_hex = 20.0/dimen
+
+    ### Geometry ###
+    cell = np.empty((dimen, dimen, dimen), dtype=object)
+    surfaces = np.empty((dimen+1, 3), dtype=object)
+
+    geometry = openmc.Geometry()
+    universe = openmc.Universe(universe_id=1, name="Contains all hexes")
+
+    for i in range(0,dimen+1):
+        surfaces[i][0] = openmc.XPlane(-10.0+i*size_hex, name="X plane at "+str(-10.0+i*size_hex))
+        surfaces[i][1] = openmc.YPlane(-10.0+i*size_hex, name="Y plane at "+str(-10.0+i*size_hex))
+        surfaces[i][2] = openmc.ZPlane(-10.0+i*size_hex, name="Z plane at "+str(-10.0+i*size_hex))
+
+        surfaces[i][0].boundary_type = 'vacuum'
+        surfaces[i][1].boundary_type = 'vacuum'
+        surfaces[i][2].boundary_type = 'vacuum'
+
+    for k in range(0,dimen):
+        for j in range(0,dimen):
+            for i in range(0,dimen):
+                cell[i][j][k] = openmc.Cell(name=("x " + str(i) +" y " + str(j) +" z " + str(k)))
+                cell[i][j][k].region = +surfaces[i][0] & -surfaces[i+1][0] & \
+                                    +surfaces[j][1] & -surfaces[j+1][1] & \
+                                    +surfaces[k][2] & -surfaces[k+1][2]
+                cell[i][j][k].fill = water_mat
+                universe.add_cell(cell[i][j][k])
+
+    geometry = openmc.Geometry(universe)
+
+    mesh_filename = "test_mesh_tets.e"
+
+    uscd_mesh = openmc.UnstructuredMesh(mesh_filename, test_opts['library'])
+
+    ### Tallies ###
+
+    # create tallies
+    tallies = openmc.Tallies()
+
+    tally1 = openmc.Tally(1)
+    tally1.scores = ['scatter', 'total', 'absorption']
+    # Export tallies
+    tallies = openmc.Tallies([tally1])
+    tallies.export_to_xml()
+
+    ### Settings ###
+    settings = openmc.Settings()
+    settings.run_mode = 'fixed source'
+    settings.particles = 5000
+    settings.batches = 2
+
+    # settings.create_fission_neutrons = False
+    settings.tracks = [(1, 1, 1)]

```

Comment:
After reading the documentation on the new tracks capability, I don't _think_ we need this line.